src/Microsoft.IdentityModel.Tokens/Validation/Validators.Audience.cs

```diff
@@ -3,12 +3,29 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
 using Microsoft.IdentityModel.Abstractions;
 using Microsoft.IdentityModel.Logging;
 
+#nullable enable
 namespace Microsoft.IdentityModel.Tokens
 {
+    /// <summary>
+    /// Definition for delegate that will validate the audiences value in a token.
+    /// </summary>
+    /// <param name="audiences">The audiences to validate.</param>
+    /// <param name="securityToken">The <see cref="SecurityToken"/> that is being validated.</param>
+    /// <param name="validationParameters"><see cref="TokenValidationParameters"/> required for validation.</param>
+    /// <param name="callContext"></param>
+    /// <returns>A <see cref="IssuerValidationResult"/>that contains the results of validating the issuer.</returns>
+    /// <remarks>This delegate is not expected to throw.</remarks>
+    internal delegate AudienceValidationResult ValidateAudience(
+        IEnumerable<string> audiences,
```

Comment:
How much work would it be to change IEnumberable<string> to IList<string>?

Reply:
Remember this call will arrive from 
```cs
TokenHandler.ValidateTokenAsync(
    string token,
    TokenValidationParameters validationParameters,
    CallContext callContextalidationParameters, 
    CancellationToken cancellationToken)
{
    // ...
}
```

Reply:
After speaking with Keegan about this, I generally agree that we should try to move to List.
However, right now we have 2 methods co-existing, one of which is public and receives an `IEnumerable`.

I am tempted to do something like:

```csharp
List<string> internalAudiences;  
if (audiences is List<string>)  
    internalAudiences = audiences;  
else  
    internalAudiences = audiences.ToList()
```

inside of the audience matching method and iterate over the `List<string>`, without breaking our public interface or  introducing the variance from `IEnumerable` to `IList` across the two methods.

Reply:
Updated to use IList on the new ValidateAudience method.